libs/scroom/src/transformpresentation.cc

```diff
@@ -114,12 +114,12 @@ bool TransformPresentation::getTransparentBackground()
   return colormappable->getTransparentBackground();
 }
 
-PipetteLayerOperations::PipetteColor TransformPresentation::getAverages(Scroom::Utils::Rectangle<int> area)
+PipetteLayerOperations::PipetteColor TransformPresentation::getPixelAverages(Scroom::Utils::Rectangle<int> area)
 {
-  PipetteViewInterface::Ptr pipetteLayer = boost::dynamic_pointer_cast<PipetteViewInterface>(presentation);
-  if(pipetteLayer)
+  PipetteViewInterface::Ptr pipettePresentation = boost::dynamic_pointer_cast<PipetteViewInterface>(presentation);
+  if(pipettePresentation)
```

Comment:
That should be `require(pipettePresentation)`: I.e. if someone calls this, then the cast must succeed (caller should inspect the correct property on the presentation to guarantee this)

There's a few other places where this can be improved.